Dockerfile

```
@@ -1,25 +1,37 @@
 FROM ruby:3.0.3-alpine
-RUN apk add gcc git libc6-compat libc-dev make nodejs postgresql13-dev sqlite-dev tzdata yarn
 
-WORKDIR /myapp
+RUN apk -U upgrade && \
+    apk add --update --no-cache gcc git libc6-compat libc-dev make nodejs \
+    postgresql13-dev sqlite-dev tzdata yarn
 
-COPY Gemfile /myapp/
-COPY Gemfile.lock /myapp/
-RUN gem update --system
-RUN bundle install
+RUN echo "Europe/London" > /etc/timezone && \
+    cp /usr/share/zoneinfo/Europe/London /etc/localtime
 
-COPY package.json /myapp/
-COPY yarn.lock /myapp/
-RUN yarn install --frozen-lockfile
-
-ADD . /myapp
 ENV RAILS_ENV=production \
     RAILS_SERVE_STATIC_FILES=yes \
     LANG=en_GB.UTF-8 \
     SECRET_KEY_BASE=TestKey
 
-RUN yarn build && yarn build:css
-RUN bundle exec rails assets:precompile
+WORKDIR /app
+
+COPY Gemfile Gemfile.lock ./
+
+RUN gem update --system && \
+    bundler -v && \
+    bundle config set no-cache 'true' && \
+    bundle config set no-binstubs 'true' && \
+    bundle install --retry=5 --jobs=4 --without=development && \
+    rm -rf /usr/local/bundle/cache
+
+COPY package.json yarn.lock ./
+
+RUN yarn install --frozen-lockfile --check-files
+
+COPY . .
+
+RUN yarn build && yarn build:css && \
+    bundle exec rails assets:precompile && \
```

I think the jsbundling and cssbundling libraries handles this for us.

https://github.com/rails/jsbundling-rails/blob/main/lib/tasks/jsbundling/build.rake
https://github.com/rails/cssbundling-rails/blob/main/lib/tasks/cssbundling/build.rake

```suggestion
RUN bundle exec rails assets:precompile && \
```

---

Interesting. I don't think it's working like that currently. I'll dig into it.

---

I'll open a follow-up PR to add `jdbundling`.